src/go/rpk/pkg/cli/cmd/debug/bundle/bundle_k8s_linux.go

```diff
@@ -221,6 +226,60 @@ func saveSingleAdminAPICalls(ctx context.Context, ps *stepParams, fs afero.Fs, c
 	}
 }
 
+// saveK8SResources will issue a GET request to the K8S API to a set of fixed
+// resources that we want to include in the bundle.
+func saveK8SResources(ctx context.Context, ps *stepParams, namespace string) step {
+	return func() error {
+		k8sCfg, err := rest.InClusterConfig()
+		if err != nil {
+			return fmt.Errorf("unable to get kubernetes cluster configuration: %v", err)
+		}
+
+		clientset, err := kubernetes.NewForConfig(k8sCfg)
+		if err != nil {
+			return fmt.Errorf("unable to create kubernetes client: %v", err)
+		}
+
+		pods, err := clientset.CoreV1().Pods(namespace).List(ctx, v1.ListOptions{})
+		if err != nil {
+			return fmt.Errorf("unable to get pods in the %q namespace: %v", namespace, err)
+		}
+		// This is a safeguard, so we don't end up saving empty request for
+		// namespace who don't have any pods.
+		if len(pods.Items) == 0 {
+			return fmt.Errorf("skipping resource collection, no pods found in the %q namespace", namespace)
+		}
```

Comment:
This chunk of code duplicates adminAddressesFromK8s (and this last conditional should probably be in that function) -- can the two be abstracted to a separate function?